src/bin/load-test/index.ts

```diff
@@ -0,0 +1,325 @@
+import { makeConfig } from 'cleaner-config'
+import minimist from 'minimist'
+import { join } from 'path'
+import pino from 'pino'
+
+import { AllEvents, asAllEvents } from '../types/shared-events'
+import {
+  endInstrument,
+  makeInstrument,
+  startInstrument
+} from '../utils/instrument'
+import { addToMetric, makeMetric } from '../utils/metric'
+import { prettyPrintObject } from '../utils/printing'
+import { makeSyncKey } from '../utils/utils'
+import { makeWorkerCluster } from '../utils/worker-cluster'
+import { asConfig, Config, configSample } from './config'
+import { WorkerConfig } from './worker'
+
+const logger = pino({
+  level: process.env.LOG_LEVEL ?? 'info'
+})
+
+// Types
+
+interface State {
+  startTime: number
+  syncKeys: string[]
+}
+
+// State:
+
+let config: Config
+
+// Metrics
+const bytesMetric = makeMetric()
+const updateTimeMetric = makeMetric()
+const readTimeMetric = makeMetric()
+
+// Instruments
+const updateTimeInstrument = makeInstrument()
+const readTimeInstrument = makeInstrument()
+
+// Initialize state
+const state: State = {
+  startTime: Date.now(),
+  syncKeys: []
+}
+
+async function main(): Promise<void> {
+  logger.info({ msg: 'starting load test', config })
+
+  // Generate an array of unique repo IDs of configured repoCount length
+  const syncKeys = generateSyncKeys(config.syncKeyPrefix, config.repoCount)
+
+  logger.debug({ msg: 'syncKeys generated', syncKeys })
+
+  // Worker Cluster Process
+  // ---------------------------------------------------------------------
+  // spawn
+  const workerCluster = makeWorkerCluster(
+    join(__dirname, 'worker'),
+    payload => {
+      try {
+        onEvent(asAllEvents(payload))
+      } catch (err) {
+        logger.error({ msg: 'worker cluster output error', err, payload })
+        exit('worker exception')
+      }
+    },
+    err => exit('worker exception', err)
+  )
+
+  // Start Inital Worker Routines
+  // ---------------------------------------------------------------------
+  syncKeys.forEach(syncKey => startWorkerRoutine(workerCluster, syncKey))
+
+  const intervalIds: NodeJS.Timeout[] = []
+
+  // Increase repo count every minute
+  function repoCountIncreaser(): void {
+    const numOfActiveRepos = state.syncKeys.length
+    const numOfActiveReposUpdated = Math.min(
+      numOfActiveRepos * config.repoCountIncreaseRatePerMin,
+      config.maxRepoCount
+    )
+    const numOfNewRepos = Math.round(numOfActiveReposUpdated - numOfActiveRepos)
+
+    if (numOfActiveRepos + numOfNewRepos > config.maxRepoCount) {
+      return
+    }
+
+    logger.debug({ msg: 'repo-increase', numOfNewRepos })
+
+    for (let i = 0; i < numOfNewRepos; ++i) {
+      const newSyncKey = makeSyncKey(
+        state.syncKeys.length,
+        config.syncKeyPrefix
+      )
+      startWorkerRoutine(workerCluster, newSyncKey)
+      logger.debug({ msg: 'new-repo', newSyncKey })
+    }
+  }
+  intervalIds.push(setInterval(repoCountIncreaser, 60000))
```

Comment:
The `makePeriodicTask` helper from `edge-server-tools` could be used here, since it adds error handling & promises, but that's up to you.